docs/dataservice-read-from-stream-layer-example.md

```diff
@@ -0,0 +1,193 @@
+
+You need to create the `OlpClientSettings` object to get catalog and partition metadata and retrieve versioned and volatile layer data from OLP.
+
+To create the `OlpClientSettings` object:
+
+1. To perform requests asynchronously, create the `TaskScheduler` object.
+
+   ```cpp
+   std::shared_ptr<olp::thread::TaskScheduler> task_scheduler =
+         olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(1u);
+   ```
+
+2. To internally operate with the OLP services, create the `Network` client.
+
+   ```cpp
+   std::shared_ptr<olp::http::Network> http_client = olp::client::
+        OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler();
+   ```
+
+   > **Note:** The `Network` client is designed and intended to be shared.
+
+3. [Authenticate](#authenticate-to-here-olp-using-client-credentials) to OLP.
+
+4. Set up the `OlpClientSettings` object.
+
+   ```cpp
+   olp::client::OlpClientSettings client_settings;
+   client_settings.authentication_settings = auth_settings;
+   client_settings.task_scheduler = std::move(task_scheduler);
+   client_settings.network_request_handler = std::move(http_client);
+   client_settings.cache =
+       olp::client::OlpClientSettingsFactory::CreateDefaultCache({});
+   ```
+
+The `OlpClientSettings` class pulls together all the settings for customization of the client library behavior.
+
+### Get Data from a Stream Layer
+
+You can read messages from a [stream layer](https://developer.here.com/olp/documentation/data-user-guide/portal/layers/layers.html#stream-layers) if you subscribe to it.
+
+To get data from the stream layer:
+
+1. Get an access key ID and access key secret. For instructions, see [Authenticate to HERE OLP Using Client Credentials](#authenticate-to-here-olp-using-client-credentials).
+
+2. Create the `OlpClientSettings` object. For instructions, see [Create OLP Client Settings](#create-olpclientsettings).
+
+3. Create the `StreamLayerClient` object with the HERE Resource Name (HRN) of the catalog that contains the layer, layer ID, and OLP client settings from step 2.
+
+   ```cpp
+   olp::dataservice::read::StreamLayerClient client(
+        client::HRN catalog, std::string layer_id,
+        client::OlpClientSettings settings);
+   ```
+
+4. Create the `SubscribeRequest` object with the `serial` or `parallel` subscription type.
+
+   - If your application should read smaller volumes of data using a single subscription, use the `serial` subscription type.
+
+   - If your application should read large volumes of data in a parallel manner, use the `parallel` subscription type and subscription ID.
+
+   ```cpp
+   auto request = olp::dataservice::read::SubscribeRequest()
+                     .WithSubscriptionMode(olp::dataservice::read::SubscribeRequest::SubscriptionMode::kSerial));
+   ```
+
+5. Call the `Subscribe` method with the `SubscribeRequest` parameter.
+
+   ```cpp
+   client::CancellationToken Subscribe(SubscribeRequest request,
```

Comment:
```suggestion
   client::CancellableFuture<SubscribeResponse> Subscribe(
      SubscribeRequest request);
```
As you also use the future based APIs below for `Poll()` and `GetData()`

Reply:
Changed